tests/by-util/test_cksum.rs

```diff
@@ -1277,3 +1277,69 @@ fn test_non_utf8_filename() {
         .stdout_is_bytes(b"SHA256 (funky\xffname) = e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\n")
         .no_stderr();
 }
+
+#[test]
+fn test_check_comment_line() {
+    // A comment in a checksum file shall be discarded unnoticed.
+
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    at.write("foo", "foo-content\n");
+    at.write(
+        "CHECKSUM-sha1",
+        "\
+        # This is a comment\n\
+        SHA1 (foo) = 058ab38dd3603703b3a7063cf95dc51a4286b6fe\n\
+        # next comment is empty\n#",
+    );
+
+    scene
+        .ucmd()
+        .arg("--check")
+        .arg("CHECKSUM-sha1")
+        .succeeds()
```

Comment:
Currently the test passes without applying the change to `checksum.rs` because `stderr` is not checked:
```suggestion
        .succeeds()
        .no_stderr()
```